src/IonTextReader.ts

```diff
@@ -89,13 +89,29 @@ export class TextReader implements Reader {
   }
 
     isIVM(input : string) : boolean {
-        if(this.depth() > 0) return false;
-        if(input.length < 6 || this.annotations().length > 0) return false;
+        if (this.depth() > 0) return false;
+        if (input.length < 6 || this.annotations().length > 0) return false;
         let prefix = "$ion_";
-        for(let i = 0; i < prefix.length; i++){
-            if(prefix.charAt(i) !== input.charAt(i)) return false;
+        let i = 0;
+
+        while (i < prefix.length) {
```

Comment:
potential simplification:  call `input.startsWith(prefix)` instead of this loop

ok if you'd prefer to leave as-is

Reply:
looks like indexOf() is about 100x faster than startswith. I'll alter all the char checks